CrossPlatformDesktopProject/CrossPlatformDesktopProject/Link/Equipables/Bomb.cs

```diff
@@ -0,0 +1,130 @@
+            } else if (current_frame == 1 || my_frame_index >= 2)
+            {
+                destination = new Rectangle((int)currentPos.X, (int)currentPos.Y, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+                destination = new Rectangle((int)currentPos.X + 48, (int)currentPos.Y, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+                destination = new Rectangle((int)currentPos.X + 24, (int)currentPos.Y + 38, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+                destination = new Rectangle((int)currentPos.X - 24, (int)currentPos.Y - 38, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+            } else if (current_frame == 2)
+            {
+                destination = new Rectangle((int)currentPos.X, (int)currentPos.Y, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+                destination = new Rectangle((int)currentPos.X - 48, (int)currentPos.Y, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+                destination = new Rectangle((int)currentPos.X + 24, (int)currentPos.Y - 38, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+                destination = new Rectangle((int)currentPos.X - 24, (int)currentPos.Y + 38, source.Width * 3, source.Height * 3);
+                spriteBatch.Draw(texture, destination, source, Color.White);
+            } 
+            
+
+        }
+
+        public void Update()
+        {
+            if (my_frame_index == 0)
+            {
+                if (++delay_frame_index >= explode_time)
+                {
+                    my_frame_index++;
+                    delay_frame_index = 0;
+                }
+                
+            } else
+            {
+                if (my_frame_index == 1)
+                {
+                    if (current_frame == 1 && ++delay_frame_index >= delay_frames)
+                    {
+                        current_frame++;
+                        delay_frame_index = 0;
+                        swap_frame_index++;
+                    } else if (current_frame == 2 && ++delay_frame_index >= delay_frames)
+                    {
+                        current_frame--;
+                        delay_frame_index = 0;
+                        swap_frame_index++;
+                    } else if (swap_frame_index >= swap_frame)
+                    {
+                        my_frame_index++;
+                        delay_frame_index = 0;
+                    }
+                } else if (my_frame_index == 2 && ++delay_frame_index >= delay_frames)

```

Comment:
This long block of nested ifs and elses is really hard to reason about, particularly when there are side effects like the ++ in multiple different the conditionals.

Reply:
It's confusing that the 4 big cases are at different indentation levels. Maybe a switch-case would be better?

Reply:
Yes, I wanted to rewrite the update logic for this - when I first wrote it, I mainly wanted to just get something in there that would make it work, then from there I planned on rewriting it using that as a template to make it better and make more sense